packages/trpc/server/routers/viewer/bookings.tsx

```diff
@@ -262,65 +496,15 @@ export const bookingsRouter = createProtectedRouter()
         );
         await Promise.all(promises);
       }
-    },
-  })
-  .middleware(async ({ ctx, rawInput, next }) => {
-    // Endpoints that just read the logged in user's data - like 'list' don't necessary have any input
-    if (!rawInput) return next({ ctx: { ...ctx, booking: null } });
-    const webhookIdAndEventTypeId = commonBookingSchema.safeParse(rawInput);
-    if (!webhookIdAndEventTypeId.success) throw new TRPCError({ code: "PARSE_ERROR" });
-
-    const { bookingId } = webhookIdAndEventTypeId.data;
-    const booking = await ctx.prisma.booking.findFirst({
-      where: {
-        id: bookingId,
-        AND: [
-          {
-            OR: [
-              /* If user is organizer */
-              { userId: ctx.user.id },
-              /* Or part of a collective booking */
-              {
-                eventType: {
-                  schedulingType: SchedulingType.COLLECTIVE,
-                  users: {
-                    some: {
-                      id: ctx.user.id,
-                    },
-                  },
-                },
-              },
-            ],
-          },
-        ],
-      },
-      include: {
-        attendees: true,
-        eventType: true,
-        destinationCalendar: true,
-        references: true,
-        user: {
-          include: {
-            destinationCalendar: true,
-            credentials: true,
-          },
-        },
-      },
-    });
-    return next({ ctx: { ...ctx, booking } });
-  })
-  .middleware(async ({ ctx, next }) => {
```

Comment:
instead of nested middlewares, we're using multiple input parsers: https://trpc.io/docs/v10/procedures#multiple-input-parsers